interpreter/hotspot/instance.go

```diff
@@ -409,7 +409,7 @@ func (d *hotspotInstance) getJITInfo(addr libpf.Address, addrCheck uint32) (
 	// scopes_pcs is a look up table to map RIP to scope_data
 	// metadata is the array that maps scope_data method indices to "class Method"
 
-	const maxMetadataSize = 4 * 1024 * 1024
+	const maxMetadataSize = 1024 * 1024
```

Comment:
Earlier this was compared against the offset from `nmethod` start, and it had to account also for the actual machine code size. Now this is this is the limit for the data to be read, and can be reduced a bit. This is still an empirically determined number. If this causes problems during testing, lets adjust it back.